torch/_inductor/scheduler.py

```diff
@@ -296,10 +298,11 @@ def decide_inplace_update(self):
             ordered_reads = sorted(self.read_writes.reads, key=lambda x: x.name)
 
             for read in ordered_reads:
-                input_node: BaseSchedulerNode = self.scheduler.name_to_node.get(
-                    read.name
-                )
+                input_node: Optional[
+                    BaseSchedulerNode
+                ] = self.scheduler.name_to_node.get(read.name)
```

Comment:
can we type `name_to_node` instead of typing this local variable? IMO we should avoid local variable types where possible because mypy can infer them

Reply:
Added type hint to `name_to_node`. I feel we still want to keep type hint of input node to improve readability.